witness/src/Dscp/Snowdrop/Expanders.hs

```diff
@@ -306,6 +325,94 @@ seqExpandersBalanceTx feesReceiverAddr (Fees minimalFees) =
     -- Expander returns account changes only
     outP = Set.fromList [accountPrefix, txPrefix, txHeadPrefix, txOfPrefix]
 
+--------------------------------------------------------------------------
+-- Block meta tx
+--------------------------------------------------------------------------
+
+-- | Expands block meta.
+-- Should happend strictly after validation of transactions in block's body.
+-- Those checks which happen in block integrity validation are missing here.
+seqExpandersBlockMetaTx
+    :: HasCoreConfig
+    => SeqExpanders Exceptions Ids Proofs Values ctx BlockMetaTx
+seqExpandersBlockMetaTx =
+    SeqExpanders $ one $ Expander inP outP $ \BlockMetaTx{..} -> do
+        let block@Block{ bHeader = header, bBody = body } = bmtBlock
+        let issuerAddr = mkAddr (hIssuer header)
+        let isGenesis = header == bHeader genesisBlock
+
+        {- Validation -}
+
+        when isGenesis $
+            -- further we assume that our genesis block has the least possible difficulty
+            unless (hDifficulty header == 0) $
+                error "Genesis block difficulty must be 0"
```

Comment:
And why should we just crash on this occasion? Maybe better to throw a decent exception instead?

Reply:
Well, it's more about programmer's mistake than about handling normal-in-the-wild case, `error` seems to be more justified (?)

Reply:
@Martoon-00 AFAIU this code is used when we are validating the block which comes from the _outside_, so it might be just anything. For me, it seems like this line of code makes any malformed genesis block with difficulty > 1 a remote killing spell for witness node.

Reply:
Ah, well, indeed, the block header should be exactly the same as the genesis header from configuration. Well yes, then it's impossible to fire malformed genesis spells, having `error` here is fine.